server/src/QRServer/db/connector.py

```diff
@@ -473,8 +481,276 @@ async def _update_or_insert_rating(self, c, user: UserRating, new_rating: float,
+        # TODO should we check if len(participants) > 0?
+        c = await self.conn.cursor()
+        now_ts = int(datetime.now().timestamp())  # current datetime as integer timestamp
+        await c.execute(
+            """
+            update tournaments
+            set start_timestamp = ?
+            where id = ? and start_timestamp is null
+            """,
+            (now_ts, tournament_id)
+        )
+        await self.conn.commit()
+
+    async def add_participant(self, tournament_id: str, user_id: str):
+        c = await self.conn.cursor()
+        await c.execute(
+            "select id from tournaments where id = ? and start_timestamp is null",
+            (tournament_id,)
+        )
+        row = await c.fetchone()
+        if not row:
+            raise ValueError("Tournament not found or already started")
+
+        tournament_id = row[0]
+
+        # Insert participant (ignore if already exists)
+        await c.execute(
+            """
+            insert or ignore into tournament_participants (tournament_id, user_id)
+            values (?, ?)
+            """,
+            (tournament_id, user_id)
+        )
+        await self.conn.commit()
+
+    async def list_participants(self, tournament_id: str) -> list[Participant]:
+        c = await self.conn.cursor()
+        await c.execute(
+            "select tournament_id, user_id from tournament_participants where tournament_id = ? order by user_id",
+            (tournament_id,)
+        )
+        rows = await c.fetchall()
+        if not rows:
+            raise ValueError("Tournament not found")
+        return [Participant(row[0], row[1]) for row in rows]
+
+    async def remove_participant(self, tournament_name, user_id):
+        c = await self.conn.cursor()
+
+        await c.execute(
+            "select id, start_timestamp from tournaments where name = ?",
+            (tournament_name,)
+        )
+        row = await c.fetchone()
+        if not row:
+            # Tournament not found
+            return
+
+        tournament_id, start_ts = row
+
+        if start_ts is not None:
+            # Tournament already began
+            return
+
+        await c.execute(
+            """
+            delete from tournament_participants
+            where tournament_id = ? and user_id = ?
+            """,
+            (tournament_id, user_id)
+        )
+        await self.conn.commit()
+
+    async def add_duel(self, tournament_id: str, duel_idx: int, active_until: datetime,
+                       user1_id: str | None, user2_id: str | None):
+        c = await self.conn.cursor()
+        await c.execute(
+            "select id from tournaments where id = ? and start_timestamp is not null and end_timestamp is null",
+            (tournament_id,)
+        )
+        row = await c.fetchone()
+        if not row:
+            raise ValueError("Tournament not found, have not started or has ended")
+
+        tournament_id = row[0]
```

Comment:
This should be removed, the database should check foreign keys

Reply:
done